ipld/share.go

```diff
@@ -47,3 +57,59 @@ func (ns NamespacedShares) Raw() [][]byte {
 	}
 	return res
 }
+
+type NamespacedShareWithProof struct {
+	ID namespace.ID
+	Share
+	Proof *nmt.Proof
+}
+
+// NewShareWithProof takes leaf and it's path, starting from the tree root,
+// and computes nmt.Proof for this leaf
+func NewShareWithProof(index int, leaf []byte, pathToLeaf []cid.Cid) *NamespacedShareWithProof {
+	rangeProofs := make([][]byte, 0)
+	for idx := len(pathToLeaf) - 1; idx >= 0; idx-- {
```

Comment:
Nit but it's better to use something like `i` as it's conventional for index inside loops.